linux/LinuxProcessList.c

```diff
@@ -472,9 +472,9 @@ static bool LinuxProcessList_readSmapsFile(LinuxProcess* process, const char* di
    ssize_t nread=0;
    int tmp=0;
    if(haveSmapsRollup) {// only available in Linux 4.14+
-      snprintf(buffer, PAGE_SIZE-1, "%s/%s/smaps_rollup", dirname, name);
+      snprintf(buffer, sizeof(buffer) - 1, "%s/%s/smaps_rollup", dirname, name);
```

Comment:
1. Should be `xSnprintf`, 2. `snprintf` already ensures terminating 0, no need for -1. 3. `buffer` on stack is uninitialized, thus -1 won't even achieve what you intend.